src/ShiftedProximalOperators.jl

```diff
@@ -156,6 +180,39 @@ the nonsmooth term is applied.
 """
 @inline prox_zero(q::R, l::R, u::R) where {R <: Real} = min(max(q, l), u)
 
+"""
+    iprox_zero(d, g, l, u)
+
+Return the solution of
+
+    min ½ d y² + g y subject to l ≤ y ≤ u
+
+for any d, g, assuming that 0 ∈ [l, u] if d < eps() and g = 0.
+This problem occurs when computing the iprox with respect to a
+separable nonsmooth term along a variable that is not part of those to which
+the nonsmooth term is applied.
+"""
+function iprox_zero(d::R, g::R, l::R, u::R) where {R <: Real}
+  if d > eps(R) 
+    argmin_quad = -g / d
+    y = min(max(argmin_quad, l), u)
+  elseif d < -eps(R)
+    d_2 = d / 2
+    val_l = d_2 * l^2 + g * l
+    val_u = d_2 * u^2 + g * u
+    y = (val_l < val_u) ? l : u
+  else # abs(d) ≤ eps(R)
+    if g > zero(R)
+      y = l
+    elseif g < zero(R)
+      y = u
+    else # g == 0
+      y = zero(R)
```

Comment:
If d = 0 and g = 0, any element of [l, u] is a solution. It's not necessary to assume that 0 ∈ [l, u].

Reply:
I assume that for our use in RegularizedOptimization, if we arrive at this given case for a component $i$ of $y$, we would want to keep $y_i = 0$ so that $x_i$ (or $x_i + s_i$ if shifted twice) stays the same at the next iteration of the used algorithm in RegularizedOptimization.

(In my previous sentence $i$ refers to an index, not an iteration number).

What would you want me to set as value instead?